tests/test_cmdline.py

```diff
@@ -871,24 +930,31 @@ def html_report(self, *args_unused, **kwargs_unused):
     def xml_report(self, *args_unused, **kwargs_unused):
         return self.xml_result
 
+    def json_report(self, *args_unused, **kwargs_unused):
+        return self.json_result
+
 
 @pytest.mark.parametrize("results, fail_under, cmd, ret", [
     # Command-line switch properly checks the result of reporting functions.
-    ((20, 30, 40), None, "report --fail-under=19", 0),
-    ((20, 30, 40), None, "report --fail-under=21", 2),
-    ((20, 30, 40), None, "html --fail-under=29", 0),
-    ((20, 30, 40), None, "html --fail-under=31", 2),
-    ((20, 30, 40), None, "xml --fail-under=39", 0),
-    ((20, 30, 40), None, "xml --fail-under=41", 2),
+    ((20, 30, 40, 50), None, "report --fail-under=19", 0),
+    ((20, 30, 40, 50), None, "report --fail-under=21", 2),
+    ((20, 30, 40, 50), None, "html --fail-under=29", 0),
+    ((20, 30, 40, 50), None, "html --fail-under=31", 2),
+    ((20, 30, 40, 50), None, "xml --fail-under=39", 0),
+    ((20, 30, 40, 50), None, "xml --fail-under=41", 2),
+    ((20, 30, 40, 50), None, "json --fail-under=49", 0),
+    ((20, 30, 40, 50), None, "json --fail-under=51", 2),
     # Configuration file setting properly checks the result of reporting.
-    ((20, 30, 40), 19, "report", 0),
-    ((20, 30, 40), 21, "report", 2),
-    ((20, 30, 40), 29, "html", 0),
-    ((20, 30, 40), 31, "html", 2),
-    ((20, 30, 40), 39, "xml", 0),
-    ((20, 30, 40), 41, "xml", 2),
+    ((20, 30, 40, 50), 19, "report", 0),
+    ((20, 30, 40, 50), 21, "report", 2),
+    ((20, 30, 40, 50), 29, "html", 0),
+    ((20, 30, 40, 50), 31, "html", 2),
+    ((20, 30, 40, 50), 39, "xml", 0),
+    ((20, 30, 40, 50), 41, "xml", 2),
+    ((20, 30, 40, 50), 49, "json", 0),
+    ((20, 30, 40, 50), 51, "json", 2),
     # Command-line overrides configuration.
-    ((20, 30, 40), 19, "report --fail-under=21", 2),
+    ((20, 30, 40, 50), 19, "report --fail-under=21", 2),
```

Comment:
Sorry about this bunch of lines.  I should come up with a better way to test this....